pkg/controller/kubelet-config/kubelet_config_nodes.go

```diff
@@ -0,0 +1,409 @@
+			if isNotFound {
+				_, err = ctrl.client.MachineconfigurationV1().MachineConfigs().Create(context.TODO(), mc, metav1.CreateOptions{})
+			} else {
+				_, err = ctrl.client.MachineconfigurationV1().MachineConfigs().Update(context.TODO(), mc, metav1.UpdateOptions{})
+			}
+			return err
+		}); err != nil {
+			return fmt.Errorf("Could not Create/Update MachineConfig, error: %v", err)
+		}
+		glog.Infof("Applied Node configuration %v on MachineConfigPool %v", key, pool.Name)
+
+		// updating the nodeConfig status based on the machineconfigpool status
+		err = ctrl.updateNodeConfigStatus(pool, nodeConfig)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (ctrl *Controller) enqueueNodeConfig(nodeConfig *osev1.Node) {
+	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(nodeConfig)
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("Couldn't get key for object %#v: %v", nodeConfig, err))
+		return
+	}
+	ctrl.nodeConfigQueue.Add(key)
+}
+
+func (ctrl *Controller) updateNodeConfig(old, cur interface{}) {
+	oldNode, ok := old.(*osev1.Node)
+	if !ok {
+		utilruntime.HandleError(fmt.Errorf("Couldn't retrieve the old object from the Update Node Config event %#v", old))
+		return
+	}
+	newNode, ok := cur.(*osev1.Node)
+	if !ok {
+		utilruntime.HandleError(fmt.Errorf("Couldn't retrieve the new object from the Update Node Config event %#v", cur))
+		return
+	}
+	if !reflect.DeepEqual(oldNode.Spec, newNode.Spec) {
+		// skipping the update in case of the Worker-Latency-Profile type transition from "Default" to "LowUpdateSlowReaction" and vice-versa
+		// (TODO) Ideally the user request has to be honoured, the transition need to be from Default -> Medium -> Low or vice-versa.
+		// Restricting the request for now until this process is automated in future.
+		if (oldNode.Spec.WorkerLatencyProfile == osev1.DefaultUpdateDefaultReaction && newNode.Spec.WorkerLatencyProfile == osev1.LowUpdateSlowReaction) || (oldNode.Spec.WorkerLatencyProfile == osev1.LowUpdateSlowReaction && newNode.Spec.WorkerLatencyProfile == osev1.DefaultUpdateDefaultReaction) {
+			message := fmt.Sprintf("Skipping the Update Node event, name: %s, transition not allowed from old WorkerLatencyProfile: %s to new WorkerLatencyProfile: %s", newNode.Name, oldNode.Spec.WorkerLatencyProfile, newNode.Spec.WorkerLatencyProfile)
+			glog.Infof(message)
+			ctrl.updateNodeConfigDegradedStatus(newNode, message, "UpdateSkipped")
+			return
+		}
+		glog.V(4).Infof("Updating the node config resource, name: %s", newNode.Name)
+		ctrl.enqueueNodeConfig(newNode)
+	}
+}
+
+func (ctrl *Controller) addNodeConfig(obj interface{}) {
+	nodeConfig, ok := obj.(*osev1.Node)
+	if !ok {
+		utilruntime.HandleError(fmt.Errorf("Couldn't retrieve the object from the Add Node Config event %#v", obj))
+		return
+	}
+	glog.V(4).Infof("Adding the node config resource, name: %s", nodeConfig.Name)
+	ctrl.enqueueNodeConfig(nodeConfig)
+}
+
+func (ctrl *Controller) deleteNodeConfig(obj interface{}) {
```

Comment:
Have you tested deleting a node config object? Does each MC correctly update? i.e. does the change propagate to the 98-xxx and 99-xxx ones?

Reply:
Yeah, this has been tested and verified. The change is getting reflected in case of the creation of kubeletconfig, featuregates.